src/components/reps/value-renderer.jsx

```diff
@@ -13,15 +12,17 @@ import arrayHandler from './array-handler'
 import dateHandler from './date-handler'
 import scalarHandler from './scalar-handler'
 import promiseHandler from './promise-handler'
+import domElementHandler from './dom-element-handler'
+import defaultHandler from './default-handler'
 
 export function renderValue(value, inContainer = false) {
   for (const handler of handlers) {
     if (handler.shouldHandle(value, inContainer)) {
       const resultElem = handler.render(value, inContainer)
-
       if (typeof resultElem === 'string') {
         return (<div>{ resultElem }</div>)
       } else if (resultElem.tagName !== undefined) {
+        // what is this even for?
```

Comment:
@mdboom @hamilton mike do you recall what this code block is for? if we don't know how it's used, let's cut it, no?

Reply:
I think this was for DOM elements before, but I believe `resultElem` almost never returns a vanilla DOM element at this point, right?

Reply:
I'll remove it with that in mind.

Reply:
This detects a DOM node and displays it verbatim.  Some of the examples in the output handling notebook use this to return custom representations of objects.  This is distinct from the cell returning a DOM element, in which case we should get a DOM inspector.  If an output handler returns a DOM element, it should be displayed verbatim (and I think this is what this does).

If we want to be careful about the recursive DOM element problem you outline above (where the user inserts a DOM element included elsewhere on the page), I think it should be possible to check if the DOM element already has a document assigned, and only do this if there isn't a document.

Reply:
Got it, will keep it in. Thanks for the clarification!